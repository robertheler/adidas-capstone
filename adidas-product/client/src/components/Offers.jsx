import React, { Component } from 'react';
import styled from 'styled-components';

const RenderDiv = styled.div`
  font-weight: 400;
  font-family: AdihausDIN,Helvetica;
  font-size: 11px;
  word-wrap: normal;
  margin: 0 0 0 0;
  color: #767677;
  word-wrap: normal;
  text-transform: uppercase;
  letter-spacing: 2px;
`;
const LearnMore = styled.a`
  color: black;
  &:hover {
    color: #fff;
    background-color: #000;
  }
  cursor: pointer;
`

const StyledOffer = styled.div`
  cursor: pointer;
  display: grid;
  grid-template-columns: 30px auto;
  &:hover {
    background: black;
    color: white;
  }
  margin: 10px;
`
const StyledPath = styled.path`
  padding: 0px;
  margin-top: 0px;
  stroke-linecap: square;
  d: path(${props => props.d});
`
const StyledDiv = styled.div`
  display: flex;
  align-items: center;
`
class Offers extends Component {
  constructor(props){
    super(props);
  }

  showModal(){
    console.log("Modal goes here");
  }
  render() {
    return (
      <RenderDiv>
        <StyledOffer>
          <StyledDiv>
            <svg height="20px" width="30px">
              <g fill="none" stroke="currentColor" strokeMiterlimit="10">
                <circle cx="9.5" cy="9.5" r="6" fill="none" stroke="currentColor"></circle>
                <circle fill="currentColor" cx="9.5" cy="6.5" r=".65"></circle>
                <StyledPath d="M9.5 8.5v4"></StyledPath>
              </g>
            </svg>
          </StyledDiv>
          <StyledDiv>WE WILL DONATE AN ADDITIONAL $2 TO THE COVID-19 SOLIDARITY RESPONSE FUND FOR WHO.</StyledDiv>
        </StyledOffer>
        <StyledOffer>
          <StyledDiv>
            <svg height="20px" width="30px">
              <g fill="none" stroke="currentColor" strokeMiterlimit="10">
                <StyledPath d="M13.42 13.5H9.5"></StyledPath>
                <StyledPath d="M4.5 5.5h10l4 3v5h-2M6.5 13.5h-2m0-6h-4"></StyledPath>
                <circle cx="8" cy="13" r="1.5"></circle>
                <circle cx="15" cy="13" r="1.5"></circle>
                <StyledPath d="M1.5 9.5h3m-2 2h2"></StyledPath>
              </g>
            </svg>
          </StyledDiv>
          <StyledDiv>FREE SHIPPING OVER $49 + FREE EXTENDED 90 DAY RETURNS & EXCHANGES.</StyledDiv>
        </StyledOffer>
        <StyledOffer>
          <StyledDiv>
            <svg height="20px" width="30px">
              <g fill="none" stroke="currentColor" strokeMiterlimit="10">

                <StyledPath d="M2.5 10.5l4 4 10-10"></StyledPath>
              </g>
            </svg>
          </StyledDiv>
          <StyledDiv>BUY NOW, PAY OVER TIME WITH PAYPAL CREDIT</StyledDiv>
        </StyledOffer>
      </RenderDiv>
    );
  }
}

export default Offers;