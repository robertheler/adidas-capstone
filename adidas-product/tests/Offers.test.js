import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Offers from '../client/src/components/Size';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0]
  };
  const wrapper = shallow(<Offers color={props.color} product={props.product} isSelected={props.isSelected} handleColorChange={props.handleColorChange}/>);
  return { wrapper, props };
}

describe('Offers Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});