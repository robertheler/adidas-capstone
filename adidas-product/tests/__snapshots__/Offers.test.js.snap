// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers Test Suite It should render correctly 1`] = `
<styled.div>
  Select Size
  <br />
  <br />
  <styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 4 / W 5",
        }
      }
      key="0"
    >
      M 4 / W 5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 1,
          "size": "M 7 / W 8",
        }
      }
      key="6"
    >
      M 7 / W 8
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 8 / W 9",
        }
      }
      key="8"
    >
      M 8 / W 9
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 9 / W 10",
        }
      }
      key="10"
    >
      M 9 / W 10
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 9.5 / W 10.5",
        }
      }
      key="11"
    >
      M 9.5 / W 10.5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 10 / W 11",
        }
      }
      key="12"
    >
      M 10 / W 11
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 10.5 / W 11.5",
        }
      }
      key="13"
    >
      M 10.5 / W 11.5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 11 / W 12",
        }
      }
      key="14"
    >
      M 11 / W 12
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 11.5 / W 12.5",
        }
      }
      key="15"
    >
      M 11.5 / W 12.5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 14,
          "size": "M 12 / W 13",
        }
      }
      key="16"
    >
      M 12 / W 13
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 3,
          "size": "M 12.5 / W 13.5",
        }
      }
      key="17"
    >
      M 12.5 / W 13.5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 13 / W 14",
        }
      }
      key="18"
    >
      M 13 / W 14
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 13.5 / W 14.5",
        }
      }
      key="19"
    >
      M 13.5 / W 14.5
    </styled.div>
    <styled.div
      inventory={
        Object {
          "quantity": 15,
          "size": "M 14 / W 15",
        }
      }
      key="20"
    >
      M 14 / W 15
    </styled.div>
  </styled.div>
</styled.div>
`;
